src/Raven.Client/Smuggler/DatabaseSmuggler.cs

```diff
@@ -1,17 +1,74 @@
+using System;
+using System.IO;
+using System.Linq;
+using System.Net.Http;
 using System.Threading.Tasks;
+using Raven.Abstractions;
+using Raven.Abstractions.Data;
 
 namespace Raven.Client.Smuggler
 {
     public class DatabaseSmuggler
     {
+        private readonly string _url;
+
+        public DatabaseSmuggler(string url)
+        {
+            _url = url;
+        }
+
         public Task ExportAsync(DatabaseSmugglerOptions options, IDatabaseSmugglerDestination destination)
         {
             throw new System.NotImplementedException();
         }
 
-        public Task ImportAsync(DatabaseSmugglerOptions options, IDatabaseSmugglerDestination destination)
+        public async Task ImportIncrementalAsync(DatabaseSmugglerOptions options,  string directoryPath)
         {
-            throw new System.NotImplementedException();
+            var files = Directory.GetFiles(directoryPath)
+                .Where(file => Constants.PeriodicExport.IncrementalExportExtension.Equals(Path.GetExtension(file), StringComparison.OrdinalIgnoreCase))
+                .OrderBy(File.GetLastWriteTimeUtc)
+                .ToArray();
+
+            if (files.Length == 0)
+                return;
+
+            // When we do incremental import, we import the indexes and transformers from the last file only, 
+            // as the previous files can hold indexes and transformers which were deleted and shouldn't be imported.
+            var oldOperateOnTypes = options.OperateOnTypes;
+            options.OperateOnTypes = options.OperateOnTypes & ~(DatabaseItemType.Indexes | DatabaseItemType.Transformers);
+            for (var i = 0; i < files.Length - 1; i++)
```

Comment:
Since we split big export files to a few parts we agreed that it make sense to also upload them in separate requests.